### DISCUSSION TITLE
bun v0.1.7

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#999] [Announcements]
To upgrade:
```bash
bun upgrade
```

To install:
```bash
curl https://bun.sh/install | bash
```

<details>
<summary>If you have any problems upgrading</summary>

Run the install script (you can run it multiple times):
```bash
curl https://bun.sh/install | bash
```
</details>

## What's new

**`bun init` quickly start a new, empty project that uses Bun** (similar to npm init). `bun init` is a new subcommand in `bun`.

<img src="https://user-images.githubusercontent.com/709451/183006613-271960a3-ff22-4f7c-83f5-5e18f684c836.gif" height=450 />

**`bun install` now supports private npm registries & scoped (authenticated) packages**

Thank you @soneymathew for your help with this.

<img width="326" alt="image" src="https://user-images.githubusercontent.com/709451/183240747-71372e23-6b34-43b5-bddf-287afef39417.png">

**`bun install` now supports lifecycle hooks for project-level package.json** (not dependencies)

<img width="445" alt="image" src="https://user-images.githubusercontent.com/709451/183240585-f5344dbf-1d9e-40ff-a9a3-f6e089c4dc77.png">

It runs postinstall scripts for your app's package.json, but ignores dependencies lifecycle hooks. This lets you use `husky`, `lint-staged`, and other postinstall-dependent packages tools 

More new stuff:
- `express` is partially supported, thanks to @zhuzilin and @evanwashere. There is a lot more work to be done - it's not fast yet and it logs a spurious error on request, but it is better than not working
-  `bun create` now lets you specify a start command so that you can say how to run the program in the output
- `process.revision` has the git sha that bun was built with

## Bug fixes
* Fix a handful of crashes that occurred in rare cases in Bun.Transpiler, `bun:ffi` and a couple other places thanks to @sno2 
* `Buffer.isBuffer` no longer checks that `this` is the `Buffer` constructor
* `Bun.Transpiler` no longer does bun-specific transforms when it shouldn't
* Make internal APIs that iterate through JS objects more reliable @sno2 in https://github.com/oven-sh/bun/pull/974
* Fix u32 jsNumber cast by @zhuzilin in https://github.com/oven-sh/bun/pull/964
* fix import in http polyfill by @zhuzilin in https://github.com/oven-sh/bun/pull/973
* fix path.normalize on ... by @zhuzilin in https://github.com/oven-sh/bun/pull/966
* allow setting status code in Response.redirect by @zhuzilin in https://github.com/oven-sh/bun/pull/985
* Fix for bearer tokens missing from request headers on bun install step by @soneymathew in https://github.com/oven-sh/bun/pull/991
* Fix of panic in threads while downloading scoped packages by @soneymathew in https://github.com/oven-sh/bun/pull/992
* feat(util): export util.TextDecoder by @xHyroM in https://github.com/oven-sh/bun/pull/990
* Fixed bugs in `latin1` and `binary` encodings in bun's `Buffer` implementation

## Misc:
* fix(makefile) fix devcontainer rule by @zhuzilin in https://github.com/oven-sh/bun/pull/957
* refactor: create a high-level property iterator by @sno2 in https://github.com/oven-sh/bun/pull/972
* fix(makefile): mkdir DEBUG_OBJ_DIR before compiling bindings by @zhuzilin in https://github.com/oven-sh/bun/pull/975
* Convert landing page to zero-JS Next.js application. by @leerob in https://github.com/oven-sh/bun/pull/945
* benchmarks(sqlite): invalid northwind database url by @xHyroM in https://github.com/oven-sh/bun/pull/989
* Update README for development help by @JL102 in https://github.com/oven-sh/bun/pull/982

## New Contributors
* @zhuzilin made their first contribution in https://github.com/oven-sh/bun/pull/964
* @leerob made their first contribution in https://github.com/oven-sh/bun/pull/945

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v0.1.6...bun-v0.1.7

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v0.1.7'>bun v0.1.7</a>.</em>

**Comment by joezak11** [comment]
after upgrading from v0.1.6 to v0.1.7 I started getting the good old `illegal hardware instruction (core dumped)` again XD

**Reply by Jarred-Sumner** [reply]
Oh no

Maybe the install script isn't detecting it properly 

Can you verify that manually downloading the -baseline build works?

**Reply by joezak11** [reply]
Nope doesn't work

**Reply by PiyushSuthar** [reply]
Having the same issue. Manually installed baseline build, still didn't work.

**Reply by Jarred-Sumner** [reply]
Can you try re-installing? Won't work for baseline yet, still figuring out what broke

**Reply by dom-inic** [reply]
bun upgrade works but its hard to know when it has completed the upgrade. seen this also with the rest of bun commands like bun install.

**Reply by Jarred-Sumner** [reply]
Can you paste a screenshot? It should print something like this 

<img width="404" alt="image" src="https://user-images.githubusercontent.com/709451/183269594-b0acc104-184c-44b7-9e67-bbe7205c0f48.png">

**Reply by dom-inic** [reply]
![Screenshot 2022-08-08 at 09 33 14](https://user-images.githubusercontent.com/47415866/183354519-12faeede-d02c-40a7-9446-2a16a612ea2a.png)
I had to force quit and check the version to see that bun has been updated.

**Reply by R0N1n-dev** [reply]
Getting illegal instruction with any bun command, including bun --help.
Tried canary and baseline and baseline canary build.


'''''IllLLLLLLLEGGGGALLLL INSTRUCTION''''''''.
Forgive the exaggerated spelling but, after having my issues closed for the same thing and being called duplicates, i feel i need to shout now to get my point across and be very clear

'ILLEGAL INSTRUCTION'
Tried 3 suggested solutions   
0.1.7 , 0.1.7 baseline and baseline canary.

**Reply by Jarred-Sumner** [reply]
After the current GitHub actions run completes, there's a good chance canary will be fixed

**Reply by PiyushSuthar** [reply]
just installed the latest canary and it worked.
![image](https://user-images.githubusercontent.com/32237558/183408995-421d2dc2-8e3a-4b21-9c41-15f133372287.png)

**Comment by akomm** [comment]
How often can we expect a raise in requirement on GLIBC? I ask, because on Ubuntu 18.04 lts you need to manually build it as 2.27 is latest package. I can upgrade to 20 lts but will I have to constantly do it to run bun? No problem for production as I use container and can set it up as I wish, but for development its annoying.

**Reply by Jarred-Sumner** [reply]
Probably once every 1-2 years.

The glibc requirement is the lowest I could figure out how to work. WebKit depends on C++ 20, but Linux distributions with glibc 2.27 and C++ 20 support are hard to find.
